app_dart/lib/src/request_handlers/dart_internal_subscription.dart

```diff
@@ -2,31 +2,208 @@
+    late Build build;
+    build = await buildBucketClient.getBuild(request);
+
+    if (build.endTime == null) {
+      log.info(
+        "Build is not finished",
+      );
+      throw UnfinishedBuildException(
+        "Build is not finished",
+      );
+    }
+    return build;
+  }
+
+  Future<Task> _createTaskFromBuildbucketResult(
+    Build build,
+    DatastoreService datastore,
+  ) async {
+    log.fine("Creating task from buildbucket result: ${build.toString()}");
+
+    final String repository = build.input!.gitilesCommit!.project!.split('/')[1];
+    log.fine("Repository: $repository");
+
+    final String branch = build.input!.gitilesCommit!.ref!.split('/')[2];
+    log.fine("Branch: $branch");
+
+    final String hash = build.input!.gitilesCommit!.hash!;
+    log.fine("Hash: $hash");
+
+    final RepositorySlug slug = RepositorySlug("flutter", repository);
+    log.fine("Slug: ${slug.toString()}");
+
+    final int startTime = build.startTime != null ? build.startTime!.millisecondsSinceEpoch : 0;
+    final int endTime = build.endTime != null ? build.endTime!.millisecondsSinceEpoch : 0;
+    log.fine("Start/end time (ms): $startTime, $endTime");
+
+    final Key<String> key = Commit.createKey(
+      db: datastore.db,
+      slug: slug,
+      gitBranch: branch,
+      sha: hash,
+    );
+
+    final String id = 'flutter/${slug.name}/$branch/$hash';
+    final Key<String> commitKey = datastore.db.emptyKey.append<String>(Commit, id: id);
+    final Commit commit = await config.db.lookupValue<Commit>(commitKey);
+    final task = Task(
+      attempts: 1,
+      buildNumber: build.number,
+      buildNumberList: build.number.toString(),
+      builderName: build.builderId.builder,
+      commitKey: key,
+      createTimestamp: startTime,
+      endTimestamp: endTime,
+      luciBucket: build.builderId.bucket,
+      name: build.builderId.builder,
+      stageName: "dart-internal",
+      startTimestamp: startTime,
+      status: _convertStatusToString(build.status!),
+      key: commit.key.append(Task),
+      timeoutInMinutes: 0,
+      reason: '',
+      requiredCapabilities: [],
+      reservedForAgentId: '',
+    );
+    return task;
+  }
+
+  Future<Task?> _getExistingTaskFromDatastore(Build build, DatastoreService datastore) async {
+    log.fine("Generating commit key from buildbucket build: ${build.toString()}");
+
+    final String repository = build.input!.gitilesCommit!.project!.split('/')[1];
+    log.fine("Repository: $repository");
+
+    final String branch = build.input!.gitilesCommit!.ref!.split('/')[2];
+    log.fine("Branch: $branch");
+
+    final String hash = build.input!.gitilesCommit!.hash!;
+    log.fine("Hash: $hash");
+
+    final RepositorySlug slug = RepositorySlug("flutter", repository);
+    log.fine("Slug: ${slug.toString()}");
+
+    final String id = 'flutter/${slug.name}/$branch/$hash';
+    final Key<String> commitKey = datastore.db.emptyKey.append<String>(Commit, id: id);
+
+    try {
```

Comment:
Why we need a try catch?

Reply:
Currently, if we call fromDatastore and it cannot find a task, it will throw an error. If we don't find an existing task, we don't want it to fail, we just want it to return null since there isn't an existing task.

Reply:
Can we add the specific Exception/Error here?

Reply:
It would also be good to log the exception/error.

Reply:
I've updated it to catch only `InternalServerError`, and log that error if it occurs!